lib/finch.ex

```diff
@@ -1,8 +1,48 @@
 defmodule Finch do
   @moduledoc """
   Finch
+
+  Create your Finch HTTP Client like so:
+
+      defmodule MyFinch do
+        use Finch
+      end
+
+  For HTTP/1 connections, Finch allows you to configure how many pools and their size,
+  for any {scheme, host, port} combination at compile time.
+
+  You can also configure a `:default` pool size and pool count that will be used for any
+  {scheme, host, port} you did not already configure.
+
+  Here is an example of what that might look like:
+
+      defmodule MyConfiguredFinch do
+        use Finch, pools: %{
+          :default => %{size: 10, count: 1},
+          {:https, "hex.pm", 443} => %{size: 20, count: 4}
+        }
+      end
   """
-  alias Finch.Conn
+  defmacro __using__(opts) do
+    quote do
+      @config %{
+        name: __MODULE__,
+        pool_sup_name: :"#{__MODULE__}.PoolSup",
+        pool_reg_name: :"#{__MODULE__}.PoolRegistry",
+        pools: Keyword.get(unquote(opts), :pools, %{})
+      }
+
+      def start_link do
+        name = :"#{__MODULE__}.PoolManager"
+        Finch.PoolManager.start_link(name, @config)
```

Comment:
What do you think about allowing the pool configuration to be passed in as part of `start_link`. That way users can set the pool sizes at runtime. I know that we would need to do this occasionally because we don't always know the location of a service at compile time. We have to get it from the environment after the app boots.

Reply:
That makes sense. How about we allow both, and I will merge the runtime config with the compile time config here?

Reply:
That seems perfect